tests/modules/similarity_functions/multiheaded_test.py

```diff
@@ -0,0 +1,35 @@
+# pylint: disable=no-self-use,invalid-name
+from numpy.testing import assert_almost_equal
+import pytest
+import torch
+from torch.nn.parameter import Parameter
+from torch.autograd import Variable
+
+from allennlp.common import Params
+from allennlp.common.checks import ConfigurationError
+from allennlp.common.testing import AllenNlpTestCase
+from allennlp.modules.similarity_functions import MultiHeadedSimilarity
+
+class TestMultiHeadedSimilarityFunction(AllenNlpTestCase):
+    def test_weights_are_correct_sizes(self):
+        # pylint: disable=protected-access
+        similarity = MultiHeadedSimilarity(num_heads=3, tensor_1_dim=9, tensor_1_projected_dim=6,
+                                           tensor_2_dim=6, tensor_2_projected_dim=12)
+        assert list(similarity._tensor_1_projection.size()) == [9, 6]
+        assert list(similarity._tensor_2_projection.size()) == [6, 12]
+        with pytest.raises(ConfigurationError):
+            similarity = MultiHeadedSimilarity(num_heads=3, tensor_1_dim=10)
+        with pytest.raises(ConfigurationError):
+            params = Params({'num_heads': 3, 'tensor_1_dim': 9, 'tensor_2_dim': 10})
+            MultiHeadedSimilarity.from_params(params)
+
+    def test_forward(self):
```

Comment:
A test for the 3D case might be nice, given it's the main use case.

Reply:
The similarity function is totally agnostic to how many dimensions there are.  I added a couple of empty dimensions to the test, but I'm not sure it really adds anything.